packages/editor/src/components/toolbar/Toolbar2.tsx

```diff
@@ -105,7 +105,11 @@ const generateToolbarMenu = () => {
 
 const toolbarMenu = generateToolbarMenu()
 
-export default function Toolbar() {
+interface Props {
+  publishSceneDisabled: boolean
+}
+
```

Comment:
If this prop is only used in the toolbar, we can directly have the [added lines](https://github.com/EtherealEngine/etherealengine/pull/10299/files#diff-1ba1511d694d6d92370f194f1133e5a389146fff09830725402671afd2988e4cR128-R131) from `Editor2Container.tsx` to `Toolbar2.tsx` file

Reply:
@aditya-mitra  Moved logic to check for publish access to the toolbar component